### PR TITLE
MINOR: Add a release script that helps generate release candidates.

### PR DESCRIPTION
**Pull request by ewencp**

### CODE REVIEW

**Comment by ewencp** [comment]
@gwenshap and @ijuma might be the most immediately interested in getting this in -- it automates a bunch of the steps in https://cwiki.apache.org/confluence/display/KAFKA/Release+Process when generating RCs. There's an incomplete branch in there for promoting RCs, partly due to lack of time and partly due to being unable to automate all of it for non-PMC members.

**Comment by ijuma** [comment]
Since @gwenshap is doing the bug fix release soon, maybe she can test drive this and merge it.

**Comment by ijuma** [comment]
By the way, this is really great. :)

**Comment by gwenshap** [comment]
Thank you!!!! Will try tomorrow morning :)

**Comment by gwenshap** [comment]
Ran into a bunch of issues related to cleanup after failure...
i.e the first iteration of running the script failed because there was a duplicate issue with fixVersion set.
I fixed this, and then it failed because it already created a branch. After removing the branch, it failed because it already created the tag.

We can decide if we want to deal with existing stuff or cleanup on exit.

**Comment by gwenshap** [comment]
More feedback:
* Of all the different things I need to manually enter or configure in advance, the base branch is hardcoded to trunk?
* I'm kinda tired of typing or copy-pasting same things over and over again. I know it is a bit much, but I'd love it to save what I entered somewhere so I wouldn't need to keep telling it where my two java homes are.

File: release.py
Comment (ijuma) [inline release.py]: Trivial fix, but it would be good to say 2.12 instead of 2.12.1 here so that we get the latest patch release. There's one other place in the file that also needs to be updated.

**Comment by ijuma** [comment]
We should close https://issues.apache.org/jira/browse/KAFKA-4421 once this PR is merged.

@gwenshap @ewencp, it may be worth agreeing on what we want to address before merging (if anything) versus what can be addressed subsequently. We should file a JIRA for unaddressed feedback and the next release manager should perhaps try to address them.

**Comment by ewencp** [comment]
@ijuma @gwenshap I've updated the script to address the concerns mentioned:

* Preferences get loaded/saved to a file once you've entered them all and you'll see them printed with the initial prompt.
* Fix cleanup. I had this partly in place but forgot to enable it for most paths because having it off was useful during debugging. It's not perfect, but captures most cases.
* Get rid of references to trunk, just store the branch the user is on when they start
* Use 2.12 instead of more specific versions.

File: release.py
Comment (gwenshap) [inline release.py]: Since this isn't implemented, perhaps lets not mention it? I found it confusing

File: release.py
Comment (gwenshap) [inline release.py]: Error message? Say "not implemented"? Something?

**Review by gwenshap — CHANGES_REQUESTED** [review]
LGTM, pending the promotion thing...

**Comment by ewencp** [comment]
@gwenshap Fair point, updated as suggested.

**Comment by ewencp** [comment]
By the way, I'm not even sure we can automate promotion. Some bits of it probably, but when I did it, it was a pretty annoying back and forth getting PMC members to do some of the steps.

**Comment by gwenshap** [comment]
LGTM

**Pull request closed without merging**
